test/e2e/run_networking_test.go

```diff
@@ -283,6 +285,35 @@ var _ = Describe("Podman run networking", func() {
 		Expect(inspectOut[0].NetworkSettings.Ports["80/tcp"][0].HostIP).To(Equal(""))
 	})
 
+	It("podman run --publish-all with EXPOSE port ranges in Dockerfile", func() {
+		podmanTest.AddImageToRWStore(ALPINE)
+		dockerfile := fmt.Sprintf(`FROM %s
+EXPOSE 20001
+EXPOSE 20000-2003
+EXPOSE 20004-2005/tcp`, ALPINE)
+		imageName := "testImage"
+		dockerfilePath := filepath.Join(podmanTest.TempDir, "Dockerfile")
+		err := ioutil.WriteFile(dockerfilePath, []byte(dockerfile), 0755)
+		Expect(err).To(BeNil())
+		podmanTest.BuildImage(dockerfilePath, imageName, "false")
+
+		// Verify that OS and Arch are being set
+		inspect := podmanTest.Podman([]string{"inspect", imageName})
+		inspect.WaitWithDefaultTimeout()
+		image := inspect.InspectImageJSON()
+		Expect(len(image[0].Config.ExposedPorts)).To(Equal(6))
+		Expect(image[0].Config.ExposedPorts).To(HaveKey("20000/tcp"))
+		Expect(image[0].Config.ExposedPorts).To(HaveKey("20001/tcp"))
+		Expect(image[0].Config.ExposedPorts).To(HaveKey("20002/tcp"))
+		Expect(image[0].Config.ExposedPorts).To(HaveKey("20003/tcp"))
+		Expect(image[0].Config.ExposedPorts).To(HaveKey("20004/tcp"))
+		Expect(image[0].Config.ExposedPorts).To(HaveKey("20005/tcp"))
```

Comment:
I think that the keys are stored as range. So you have `20001/tcp`, `20000-20003/tcp` and `20004-20005/tcp`

Reply:
You are right. Reworking.